TST: numpy dev failures for visualization

Example log: https://travis-ci.org/astropy/astropy/jobs/442639772

Snippets:
```
astropy/visualization/scripts/tests/test_fits2bitmap.py
astropy/visualization/tests/test_norm.py
astropy/visualization/wcsaxes/tests/test_frame.py
astropy/visualization/wcsaxes/tests/test_images.py
E       DeprecationWarning: np.asscalar(a) is deprecated since NumPy v1.16, use a.item() instead
```
```
astropy/visualization/wcsaxes/tests/test_images.py
E       ValueError: need at least one array to concatenate
```

cc @astrofrog @mhvk

--- a/.travis.yml
+++ b/.travis.yml
@@ -135,6 +135,7 @@ matrix:
         - os: linux
           env: NUMPY_VERSION=dev SETUP_CMD='test --remote-data'
                CONDA_DEPENDENCIES=$CONDA_ALL_DEPENDENCIES
+               MATPLOTLIB_VERSION=dev
 
         # We check numpy-dev also in a job that only runs from cron, so that
         # we can spot issues sooner. We do not use remote data here, since
@@ -148,6 +149,7 @@ matrix:
       - os: linux
         env: NUMPY_VERSION=dev SETUP_CMD='test --remote-data'
              CONDA_DEPENDENCIES=$CONDA_ALL_DEPENDENCIES
+             MATPLOTLIB_VERSION=dev
 
 install:
     - git clone git://github.com/astropy/ci-helpers.git

